rpc_component/cli.py

```diff
@@ -242,17 +238,33 @@ def dependency(components_dir, **kwargs):
         )
 
 
-def metadata(components_dir, **kwargs):
-    metadata_dir = kwargs.pop("metadata_dir")
+def get_dependents(component_name, download_dir, textflag, releases_dir, components_dir):
+    dependents = []
+    components = c_lib.load_all_components(components_dir, releases_dir)
+    for component in components:
+        requirements = [{"name": component.name, "repo_url": component.repo_url, "sha": 0}] #0 = latest commit, or use
```

Comment:
I'm not sure if I understand what you mean by "calling a component a requirement".  I thought #L245 was creating a requirement to specify which repo/component to clone.  Perhaps my confusion is with the "download_requirements" function, or the function name.  For example, reading the name "download_requirements" makes me think of downloading requirements. But I would summarize the function as "point-to/clone/fetch a repo and reset HEAD to the specified SHA", which - off the top of my head - might be appropriately named something like "get_repo_at_SHA", but since "requirements" is an input, then maybe "get_repo_based_on_requirements".  Hmm, perhaps I should just keep "download_requirements" as it was and I'll create a new function named "get_repo_based_on_requirements" (unless that's not an okay name) that does the same thing as `download_requirements` but also
checks the requirement list for a series/branch as well.

Reply:
Requirements in this project refer to specific component versions calculated based on the defined constraints. It's called `download_requirements` because it takes a [list of requirements](https://github.com/rcbops/rpc_component/blob/master/rpc_component/schemata.py#L193-L202) and downloads them. Naming things is never easy but I don't think it is misleading to use the word "download" in this context.

Downloading a Git repository and checking out a particular version is the generalised use case here. I'm happy for there to be refactoring if necessary, I just want to make sure code isn't shoehorned together.